backend/src/core/middlewares/unsubscriber.middleware.ts

```diff
@@ -38,67 +42,35 @@ const isUnsubscribeRequestValid = async (
   }
 }
 
-/**
- * Retrieves the unsubscribe status and the associated campaign
- * @param _req
- * @param res
- */
-const getUnsubscriberStatus = async (
-  _req: Request,
-  res: Response
-): Promise<Response> => {
-  const { campaign, recipient } = res.locals.unsubscribe
-  const unsubscribe = await UnsubscriberService.getUnsubscriber(
-    campaign.id,
-    recipient
-  )
-
-  return res.json({
-    unsubscribed: unsubscribe !== null,
-    campaign: {
-      name: campaign.name,
-      channelType: campaign.type,
-    },
-  })
-}
-
 /**
  * Creates a new unsubscribe record
  * @param _req
  * @param res
  * @param next
  */
-const createUnsubscriber = async (
+const findOrCreateUnsubscriber = async (
   _req: Request,
-  res: Response,
-  next: NextFunction
+  res: Response
 ): Promise<Response | void> => {
-  try {
-    const { campaign, recipient } = res.locals.unsubscribe
-    await UnsubscriberService.createUnsubscriber({
-      campaignId: campaign.id,
-      recipient,
-    })
-    return res.status(201).json({
-      unsubscribed: true,
-      campaign: {
-        name: campaign.name,
-        channelType: campaign.type,
-      },
-    })
-  } catch (err) {
-    if (err instanceof UniqueConstraintError) {
-      return res.status(400).json({
-        message: 'Campaign is already unsubscribed',
-      })
-    }
+  const { campaignId, recipient } = res.locals.unsubscriber
+  const [
+    unsubscriber,
+    created,
+  ] = await UnsubscriberService.findOrCreateUnsubscriber({
+    campaignId,
+    recipient,
+  })
 
-    next(err)
-  }
+  const statusCode = created ? 201 : 200
+  const { createdAt } = unsubscriber
+  return res.status(statusCode).json({
+    campaign_id: campaignId,
+    recipient,
+    created_at: createdAt,
```

Comment:
Hmmm how is created_at going to be used on the frontend?

Reply:
Don't think it will be used in the frontend. Let me drop that from the response. Thanks for pointing this out!